asset-transfer-basic/chaincode-java/SCRATCH.md

```diff
@@ -0,0 +1,49 @@
+
+
+
+# Scratch notes - Ignore - ... `fabric-cli` redux
+
+`fabric [options] peer <group> <command> [parameters]`
+
+```
+fabric  => network 
+peer    => implicit (from env/context)
+channel => implicit (from env/context)
+group   => chaincode 
+[params] => --param=value or NETWORK_$GROUP_$COMMAND_$PARAM=value from env 
+```
+
+```shell
+network chaincode package       <folder-path> <bundle-path> 
+network chaincode id            <bundle-path> 
+network chaincode install       <bundle-path> 
+network chaincode approve       <name> <id> 
+network chainocde commit        <name> 
+```
```

Comment:
It's fascinating and seems useful that the `network`can do all steps.
but, I cannot asure that this pattern doesn't obey SRP.

appreciated if you give any comments

Reply:
I think we can use `./network` as an approximation to the functional patterns for a `fabric` CLI.

There was some work on a [fabric-cli](https://github.com/hyperledger/fabric-cli) a few years ago, but it stalled.

Each of the different routines in the `network` script are really part of a _command group_, i.e.. `chaincode` or `channel` or ... 

I think a really good way to carry this effort forward is to construct a golang `fabric-config` SDK, with individual modules or packages that understand the various functional command groups.  The routines should ONLY work with the gRPC and https endpoints, and introduce a better organization for _Identity_ and _Context_ of an API call.

If we have a solid golang module that can administer channels, chaincode, enrollments, etc., this can be incorporated into a `chaincode` CLI binary, running as a sub-command of a higher-level `fabric` CLI driver.

More importantly, the golang module should not ONLY serve the CLI consumers, but should also serve as the basis for software automation, such as Kubernetes Operators for Fabric.

With the SRP principal, think of `chaincode`, `channel`, `context`, etc... as the utility.  Ideally the `/network` (or `fabric` / `fabctl`) routine is a simple switch to dispatch to the correct sub-command.